addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -61,175 +69,461 @@
-        dptTypeMap.put(DPTXlatorBoolean.DPT_BOOL.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_ENABLE.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_RAMP.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_ALARM.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_BINARYVALUE.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_STEP.getID(), OnOffType.class);
+        @SuppressWarnings("unused")
+        final List<Class<?>> xlators = Arrays.<Class<?>> asList(DPTXlator1BitControlled.class,
+                DPTXlator2ByteFloat.class, DPTXlator2ByteUnsigned.class, DPTXlator3BitControlled.class,
+                DPTXlator4ByteFloat.class, DPTXlator4ByteSigned.class, DPTXlator4ByteUnsigned.class,
+                DPTXlator64BitSigned.class, DPTXlator8BitSigned.class, DPTXlator8BitUnsigned.class,
+                DPTXlatorBoolean.class, DPTXlatorDate.class, DPTXlatorDateTime.class, DPTXlatorRGB.class,
+                DPTXlatorSceneControl.class, DPTXlatorSceneNumber.class, DPTXlatorString.class, DPTXlatorTime.class,
+                DPTXlatorUtf8.class
+        // calimero 2.4 DPTXlator8BitEnum.class,
+        // calimero 2.4 DptXlator8BitSet.class,
+        // calimero 2.4 DptXlatorMeteringValue.class
+        );
+
+        dptTypeMap = new HashMap<String, Class<? extends Type>>();
+        dptMainTypeMap = new HashMap<Integer, Class<? extends Type>>();
+
+        /**
+         * MainType: 1
+         * 1.005: Alarm, values no alarm alarm
+         * 1.016: Acknowledge, values no action acknowledge
+         * 1.006: Binary value, values low high
+         * 1.017: Trigger, values trigger trigger
+         * 1.007: Step, values decrease increase
+         * 1.018: Occupancy, values not occupied occupied
+         * 1.008: Up/Down, values up down
+         * 1.019: Window/Door, values closed open
+         * 1.009: Open/Close, values open close
+         * 1.010: Start, values stop start
+         * 1.021: Logical function, values OR AND
+         * 1.011: State, values inactive active
+         * 1.022: Scene A/B, values scene A scene B
+         * 1.001: Switch, values off on
+         * 1.012: Invert, values not inverted inverted
+         * 1.023: Shutter/Blinds mode, values only move up/down move up/down + step-stop
+         * 1.100: Heat/Cool, values cooling heating
+         * 1.002: Boolean, values false true
+         * 1.013: Dim send-style, values start/stop cyclic
+         * 1.003: Enable, values disable enable
+         * 1.014: Input source, values fixed calculated
+         * 1.004: Ramp, values no ramp ramp
+         * 1.015: Reset, values no action reset
+         */
+        dptMainTypeMap.put(1, OnOffType.class);
+        /** Exceptions Datapoint Types "B1", Main number 1 */
         dptTypeMap.put(DPTXlatorBoolean.DPT_UPDOWN.getID(), UpDownType.class);
```

Comment:
>dptMainTypeMap.put(1, OnOffType.class); 

responds generic inclusive (OnOffType). But there are datatypes that do not fit into that pattern(s). Exceptions which we have to put specially.